src/Umbraco.Core/IO/IOHelper.cs

```diff
@@ -188,5 +188,62 @@ public string GetRelativePath(string path)
             return PathUtility.EnsurePathIsApplicationRootPrefixed(path);
         }
 
+        /// <summary>
+        /// Retrieves array of temporary folders from the hosting environment.
+        /// </summary>
+        /// <returns>Array of <see cref="DirectoryInfo"/> instances.</returns>
+        public DirectoryInfo[] GetTempFolders()
+        {
+            var tempFolderPaths = new[]
+            {
+                _hostingEnvironment.MapPathContentRoot(Constants.SystemDirectories.TempFileUploads)
+            };
+
+            foreach (var tempFolderPath in tempFolderPaths)
+            {
+                // Ensure it exists
+                Directory.CreateDirectory(tempFolderPath);
+            }
+
+            return tempFolderPaths.Select(x => new DirectoryInfo(x)).ToArray();
+        }
+
+        /// <summary>
+        /// Cleans contents of a folder.
+        /// </summary>
+        /// <param name="folder">Folder to clean.</param>
+        /// <param name="age">Age of files within folder to delete.</param>
+        /// <returns>Result of operation</returns>
```

Comment:
Maybe it should be mentioned, that on failure, some files can still have been cleaned.

Reply:
Have updated comments to reflect this.